util/dashboard/gen_dashboard_entry.py

```diff
@@ -205,7 +204,23 @@ def get_development_stage(obj, rev, is_latest_rev=True):
 
 
 # Create dashboard of hardware IP development status
-def gen_dashboard_html(hjson_path, outfile):
+def gen_dashboard_html(config: Union[Path, Tuple[Path, str]], outfile):
+    """Generate a html table row for a given configuration.
+
+    The function takes either the path to a block configuratin file
+    or a tuple of this path plus the path to the ip's documentation directory.
+
+    This second option is for some generated hardware blocks, such as `clkmgr`,
+    which don't have their configuration file within the directory
+    containing their documentation
+    (`hw/top_earlgrey/ip/clkmgr/data/autogen/clkmgr.hjson` and `hw/ip/clkmgr/` in this case).
```

Comment:
Nit: rewrap this comment? Just eyeballing it, I'm not sure whether it exceeds 80 columns.

Reply:
I've used Harry's suggested docstring. Does it look better to your eye?

*(Interestingly the [python style guide](https://opentitan.org/book/doc/contributing/style_guides/python_coding_style.html#line-length) and `.flake8` disagree on the allowed line length)*